### PR TITLE
Expand our user-agent some more

### PR DESCRIPTION
**Pull request by stgraber**

### CODE REVIEW

File: lxd/daemon.go [hunk 1]
```diff
@@ -549,37 +553,35 @@ func (d *Daemon) init() error {
...
 	err = d.setupMAASController(maasAPIURL, maasAPIKey, maasMachine)
 	if err != nil {
 		return err
 	}
 
-	if !d.os.MockMode {
-		/* Start the scheduler */
-		go deviceEventListener(d.State())
-		readSavedClientCAList(d)
-	}
```
Comment (brauner) [inline lxd/daemon.go]: Huh, when did that get in.

File: lxd/devices.go [hunk 2]
```diff
@@ -817,7 +817,7 @@ func deviceUSBEvent(s *state.State, usb usbDevice) {
 func deviceEventListener(s *state.State) {
 	chNetlinkCPU, chNetlinkNetwork, chUSB, err := deviceNetlinkListener()
 	if err != nil {
-		logger.Errorf("scheduler: couldn't setup netlink listener")
+		logger.Errorf("scheduler: couldn't setup netlink listener: %v", err)
```
Comment (brauner) [inline lxd/devices.go]: nit and not necessary but probably be clearer if: "Couldn't set up netlink listener".

File: lxd/daemon.go [hunk 3]
```diff
@@ -518,27 +518,31 @@ func (d *Daemon) init() error {
 	/* Log expiry */
 	d.tasks.Add(expireLogsTask(d.State()))
 
-	/* set the initial proxy function and external auth based on config values in the DB */
+	/* Setup the proxy handler, external authentication and MAAS */
```
Comment (brauner) [inline lxd/daemon.go]: While you're changing this should probably be using `//` comments.

File: shared/version/useragent.go [hunk 4]
```diff
@@ -9,9 +9,11 @@ import (
 )
 
 // UserAgent contains a string suitable as a user-agent
-var UserAgent = getUserAgent(nil)
+var UserAgent = getUserAgent()
+var userAgentStorageBackends []string
```
Comment (brauner) [inline shared/version/useragent.go]: I'm likely just missing context: Where is that initialized?

Comment (brauner) [reply shared/version/useragent.go]: Ah, `backends` below. Don't see where that belongs to but ok. :)

File: lxd/daemon.go (hunk 1 above)
Comment (stgraber) [reply lxd/daemon.go]: Probably a bad merge in the clustering branch.

File: lxd/devices.go (hunk 2 above)
Comment (stgraber) [reply lxd/devices.go]: There's all wrongly capitalized so I left it for later.

File: lxd/daemon.go (hunk 3 above)
Comment (stgraber) [reply lxd/daemon.go]: Same as the other, probably best to do in one shot in a separate commit.

File: shared/version/useragent.go (hunk 4 above)
Comment (stgraber) [reply shared/version/useragent.go]: yeah, those vars each have matching public functions to set them, those functions also refresh the actual string.

**Pull request merged**